lookout/style/format/annotations/annotations.py

```diff
@@ -0,0 +1,196 @@
+"""Annotations for style-analyzer."""
+from typing import FrozenSet, Optional, Tuple
+
+from lookout.style.format.utils import get_uast_parents
+
+
+class Annotation:
+    """Base class for annotation."""
+
+    def __init__(self, start: int, stop: int):
+        """
+        Initialization.
+
+        :param start: Annotation start.
+        :param stop: Annotation end.
+        """
+        self._range = (start, stop)
+        self._start = start
+        self._stop = stop
+
+    start = property(lambda self: self._start)
+
+    stop = property(lambda self: self._stop)
+
+    range = property(lambda self: self._range)
+
+    name = property(lambda self: type(self).__name__)
+
+    def __repr__(self):
+        return self.__str__()
+
+    def __str__(self):
+        return "%s[%d, %d)" % (self.name, self.start, self.stop)
+
+
+class LineAnnotation(Annotation):
+    """Line number annotation."""
+
+    def __init__(self, start: int, stop: int, number: int):
+        """Init."""
+        super().__init__(start, stop)
+        self._number = number
```

Comment:
check for a crazy line number; we should mention in the docstring that we are either 0-based or 1-based.